src/python/pants/option/config_test.py

```diff
@@ -22,72 +38,175 @@ def setUp(self) -> None:
       disclaimer:
         Let it be known
         that.
-      blank_section:
 
       [a]
       list: [1, 2, 3, %(answer)s]
-      listappend: +[7, 8, 9]
+      list2: +[7, 8, 9]
 
       [b]
       preempt: True
+      
+      [b.nested]
```

Comment:
Nested sections like this trip up a naive TOML implementation because TOML (and JSON/YAML) represent the entire document as one single dictionary. So, this is represented as `{"b": {"nested": {"dict": "..."}}}`. So, with TOML, it's not enough to only look at the top-level `toml_values.keys()`. 

This change ensures we are robustly supporting subscopes.